application_sandbox/extended_dynamic_state2/main.cpp

```diff
@@ -0,0 +1,412 @@
+                -0.5f, -0.5f, -0.5f, 0.5f,  0.5f,  -0.5f, -0.5f, 0.5f},
+               36,
+               {0, 1, 3, 3, 2, 0, 1, 0, 6, 6, 7, 1, 4, 5, 7, 7, 6, 4,
+                2, 3, 5, 5, 4, 2, 0, 2, 4, 4, 6, 0, 3, 1, 7, 7, 5, 3}};
+
+// green cube
+const struct {
+  size_t num_vertices;
+  float positions[24];
+  float uv[16];
+  float normals[24];
+  size_t num_indices;
+  uint32_t indices[36];
+} cube_data2 = {8,
+                {1.0f,  0.5f,  0.5f,  1.0f,  -0.5f, 0.5f,  1.0f,  0.5f,
+                 -0.5f, 1.0f,  -0.5f, -0.5f, -1.0f, 0.5f,  -0.5f, -1.0f,
+                 -0.5f, -0.5f, -1.0f, 0.5f,  0.5f,  -1.0f, -0.5f, 0.5f},
+                {0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f,
+                 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f},
+                {0.5f,  0.5f,  0.5f,  0.5f,  -0.5f, 0.5f,  0.5f,  0.5f,
+                 -0.5f, 0.5f,  -0.5f, -0.5f, -0.5f, 0.5f,  -0.5f, -0.5f,
+                 -0.5f, -0.5f, -0.5f, 0.5f,  0.5f,  -0.5f, -0.5f, 0.5f},
+                36,
+                {0, 1, 3, 3, 2, 0, 1, 0, 6, 6, 7, 1, 4, 5, 7, 7, 6, 4,
+                 2, 3, 5, 5, 4, 2, 0, 2, 4, 4, 6, 0, 3, 1, 7, 7, 5, 3}};
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+struct CubeFrameData {
+  containers::unique_ptr<vulkan::VkCommandBuffer> command_buffer_;
+  containers::unique_ptr<vulkan::VkFramebuffer> framebuffer_;
+  containers::unique_ptr<vulkan::DescriptorSet> cube_descriptor_set_;
+};
+
+VkPhysicalDeviceExtendedDynamicState2FeaturesEXT kFeat{
+    VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_2_FEATURES_EXT,
+    nullptr,
+    VK_TRUE,
+    VK_FALSE,
+    VK_FALSE,
+};
+
+// This creates an application with 16MB of image memory, and defaults
+// for host, and device buffer sizes.
+class CubeSample : public sample_application::Sample<CubeFrameData> {
+ public:
+  CubeSample(const entry::EntryData* data)
+      : data_(data),
+        cubes_{
+            vulkan::VulkanModel(data->allocator(), data->logger(), cube_data),
+            vulkan::VulkanModel(data->allocator(), data->logger(), cube_data2)},
+        Sample<CubeFrameData>(
+            data->allocator(), data, 1, 512, 1, 1,
+            sample_application::SampleOptions()
+                .EnableMultisampling()
+                .EnableDepthBuffer()
+                .SetVulkanApiVersion(VK_API_VERSION_1_1)
+                .AddDeviceExtensionStructure(&kFeat),
+            {}, {VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME},
+            {VK_EXT_EXTENDED_DYNAMIC_STATE_2_EXTENSION_NAME}) {}
+
+  virtual void InitializeApplicationData(
+      vulkan::VkCommandBuffer* initialization_buffer,
+      size_t num_swapchain_images) override {
+    VkPhysicalDeviceExtendedDynamicState2FeaturesEXT ext_features{
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_2_FEATURES_EXT,
+        nullptr,
+        VK_TRUE,
```

Comment:
I would leave these all as VK_FALSE, as we want to the vkGetPhysicalDeviceFeatures2KHR call to explicitly flip it to true. The check below will pass if the call does nothing as its starts as VK_TRUE.